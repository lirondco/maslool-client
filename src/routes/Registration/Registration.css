.registrationpage {
  margin: 2%;
  padding: 2%;
}

.registrationpage h2 {
  margin-bottom: 1%;
}

hr {
  border: 1px solid;
}
