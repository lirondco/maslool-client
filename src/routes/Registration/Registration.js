import React, { Component } from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'
import './Registration.css'

export default class Registration extends Component {
    state = {
        isLoading: false
    }

    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        },
    }

    handleIsLoading = () => {
        this.setState({ isLoading: true })
    }

    handleRegistrationSuccess = () => {
        this.setState({ isLoading: false })
        const { location, history } = this.props
        const destination = (location.state || {}).from || '/welcome'
        history.push(destination)
    }

    render() {
        return (
            <section className = 'registrationpage'>
                <h2>Register</h2>
                <hr />
                <RegistrationForm 
                    onRegistrationSuccess={this.handleRegistrationSuccess}
                />
            </section>
        )
    }
}