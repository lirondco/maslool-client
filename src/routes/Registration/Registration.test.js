import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Registration from './Registration'

describe(`Regisration route`, () => {
  it('renders the complete page', () => {
    const wrapper = shallow(<Registration />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
