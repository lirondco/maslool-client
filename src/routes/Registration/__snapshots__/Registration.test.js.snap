// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regisration route renders the complete page 1`] = `
<section
  className="registrationpage"
>
  <RegistrationForm
    onRegistrationSuccess={[Function]}
  />
</section>
`;
